src/components/SignupForm.jsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { signup } from '../actions/actionCreators';
+
+const mapStateToProps = store => ({
+  games: store.games,
+});
+
+const mapDispatchToProps = dispatch => ({
+  signIn: (username, password, firstname, lastname, email) => {
+    dispatch(signup(username, password, firstname, lastname, email));
+  },
+});
+class SignupForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '', password: '', firstname: '', lastname: '', email: '',
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    if (name === 'username') this.setState({ username: event.target.value });
```

Comment:
Since you de-structured the value on line 25, you  dont need to do it again in the event object in each case.  We can also refactor this to a switch statement which will be a bit cleaner

Reply:
done